wic/src/test/java/edu/asu/diging/wic/core/service/impl/ConceptTextServiceTest.java

```diff
@@ -0,0 +1,144 @@
+package edu.asu.diging.wic.core.service.impl;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+import edu.asu.diging.wic.core.model.impl.ConceptText;
+import edu.asu.diging.wic.core.repository.ConceptTextDatabaseRepository;
+import edu.asu.diging.wic.core.service.impl.ConceptTextService;
+
+public class ConceptTextServiceTest {
+
+    @Mock
+    private ConceptTextDatabaseRepository conceptTextDatabaseConnection;
+    
+    @InjectMocks
+    private ConceptTextService conceptTextService;
+    
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+    }
+    @Test
+    public void testAddText() {
+        
+        ConceptText conceptText = new ConceptText();
+        conceptText.setAuthor("Smit");
+        conceptText.setConceptId("WC123");
+        conceptText.setText("The great Scientist");
+        conceptText.setTitle("Information");
+
+        when(conceptTextDatabaseConnection.save(conceptText)).thenReturn(conceptText);
+        ConceptText savedText = conceptTextService.addText(conceptText, "admin");
+        
+        Assert.assertEquals(conceptText, savedText);
+        Assert.assertEquals(conceptText.getAddedBy(), "admin");
+        Assert.assertNotNull(conceptText.getAddedOn());
+        
+        verify(conceptTextDatabaseConnection).save(conceptText);
+    }
+
+    @Test
+    public void testFindAll() {
+        
+        List<ConceptText> dataObj = new ArrayList<>();
+        dataObj.add(new ConceptText());
+        dataObj.add(new ConceptText());
+        
+        Page<ConceptText> dataFromDb = new PageImpl<ConceptText>(dataObj);
+        when(conceptTextDatabaseConnection.findAll(any(Pageable.class)))
+            .thenReturn(dataFromDb);
+        
+        List<ConceptText> objFromDb = conceptTextService.findAll(1, 2);
+        Assert.assertEquals(objFromDb.size(), 2);
+        verify(conceptTextDatabaseConnection).findAll(any(Pageable.class));
```

Comment:
that doesn't really need to be test, but you do need to test if you get the objects you expect to get.